build/build-image/cross/Dockerfile

```diff
@@ -43,14 +44,18 @@ RUN apt-get update \
 
 # Use dynamic cgo linking for architectures other than amd64 for the server platforms
 # To install crossbuild essential for other architectures add the following repository.
-RUN echo "deb http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/sources.list.d/cgocrosscompiling.list \
+# TODO: crossbuild-essential-${platform} is not installed for arm64 and ppc64le. We need to find the right
+# set of repo list and dependencies to enable installing crossbuild-essential-${platform} on these architectures
+RUN echo "deb [arch=amd64] http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/sources.list.d/cgocrosscompiling.list \
   && apt-key adv --no-tty --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32 \
   && apt-get update \
   && apt-get install -y build-essential \
-  && for platform in ${KUBE_DYNAMIC_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done \
+  && ([ "$(uname -m)" = "x86_64" ] && for platform in ${KUBE_DYNAMIC_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done || true) \
```

Comment:
does this uname trick work if you're running this under qemu?

(also, don't we need to install qemu if building these images for arm or ppc on amd64?)

Reply:
Didn't seem to be a problem, tried on MacOS, Ubuntu-x86, Ubuntu-arm64. Please let me know if you see a problem

Reply:
I don't see copying the qemu binary anywhere in the flow! I really doubt whether we will be able to cross-build the docker image for other archs on linux/amd64 platform.

Cross build may happen on MacOS without issues because macos will support by default(more information - https://docs.docker.com/docker-for-mac/multi-arch/)

Reply:
I'm confused by this too.. we just don't need these packages if we aren't on a x86_64 machine?